test/test/nativemem/nativemem_known_lib_crash.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "asprof.h"
+#include <dlfcn.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define ASSERT_NO_DLERROR(sym)            \
+    if (sym == NULL) {                    \
+        err = dlerror();                  \
+        if (err != NULL) {                \
+            fprintf(stderr, "%s\n", err); \
+            exit(1);                      \
+        }                                 \
+    }                                     \
+
+#define ASSERT_NO_ASPROF_ERR(err)                       \
+    if (err != NULL) {                                  \
+        fprintf(stderr, "%s\n", asprof_error_str(err)); \
+        exit(1);                                        \
+    }
+
+void outputCallback(const char* buffer, size_t size) {
+    fwrite(buffer, sizeof(char), size, stdout);
+}
+
+/*
+Idea of the test (the behavior applies as of 0c72a8d):
+- We load libcallsmalloc.so
+- We start AP without nativemem mode
+    => 'malloc' is not hooked
+- We dlclose libcallsmalloc.so
```

Comment:
A step missing before this: Stop AP.

Reply:
cda6051cebf575254e2e1d5a56afd4ba11cd11e2